hawk/app/assets/javascripts/module/primitives.js

```diff
@@ -2,300 +2,391 @@
-        title: __('Primitive ID'),
-        sortable: true,
-        switchable: false,
-        clickToSelect: true
-      }, {
-        field: 'operate',
-        title: __('Operations'),
-        sortable: false,
-        clickToSelect: false,
-        class: 'col-sm-2',
-        events: {
-          'click .delete': function (e, value, row, index) {
-            e.preventDefault();
-            var $self = $(this);
-
-            try {
-              answer = confirm(
-                i18n.translate(
-                  'Are you sure you wish to delete %s?'
-                ).fetch(row.id)
-              );
-            } catch (e) {
-              (console.error || console.log).call(console, e.stack || e);
-            }
+  // Some craziness to avoid duplicating this stuff for both
+  // primitives and templates. The whole thing is looped over
+  // twice, once for primitive and once for template
+  var controller_types = {
+    primitive: {
+      table_selector: '#primitives #middle table.primitives',
+      cib_primitives_path: Routes.cib_primitives_path,
+      cib_primitive_path: Routes.cib_primitive_path,
+      edit_cib_primitive_path: Routes.edit_cib_primitive_path,
+      form_selector: '#primitives #middle form',
+      template_selector: '#primitive_template',
+      clazz_selector: '#primitive_clazz',
+      provider_selector: '#primitive_provider',
+      type_selector: '#primitive_type',
+    },
+    template: {
+      table_selector: '#templates #middle table.primitives',
+      cib_primitives_path: Routes.cib_templates_path,
+      cib_primitive_path: Routes.cib_template_path,
+      edit_cib_primitive_path: Routes.edit_cib_template_path,
+      form_selector: '#templates #middle form',
+      template_selector: '#template_template',
+      clazz_selector: '#template_clazz',
+      provider_selector: '#template_provider',
+      type_selector: '#template_type',
+    }
+  };
 
-            if (answer) {
-              $.ajax({
-                dataType: 'json',
-                method: 'POST',
-                data: {
-                  _method: 'delete'
-                },
-                url: Routes.cib_primitive_path(
-                  $('body').data('cib'),
-                  row.id,
-                  { format: 'json' }
-                ),
-
-                success: function(data) {
-                  if (data.success) {
-                    $.growl({
-                      message: data.message
-                    },{
-                      type: 'success'
-                    });
+  $.each(controller_types, function(_, controller_type) {
+    controller_type.sel_template = "select" + controller_type.template_selector;
+    controller_type.sel_clazz = "select" + controller_type.clazz_selector;
+    controller_type.sel_provider = "select" + controller_type.provider_selector;
+    controller_type.sel_type = "select" + controller_type.type_selector;
+    controller_type.selects = [controller_type.sel_template,
```

Comment:
Identifier 'controller_type' is not in camel case.<br>Identifier 'sel_template' is not in camel case.<br>Too many errors. (9% scanned).